protocol/msgpack/aggregated_encoder.go

```diff
@@ -44,10 +44,14 @@ type aggregatedEncoder struct {
 }
 
 // NewAggregatedEncoder creates an aggregated encoder
-func NewAggregatedEncoder(encoder BufferedEncoder) AggregatedEncoder {
+func NewAggregatedEncoder(encoder BufferedEncoder, opts EncoderOptions) AggregatedEncoder {
+	if opts == nil {
+		opts = NewEncoderOptions()
+	}
+
 	enc := &aggregatedEncoder{
-		encoderBase: newBaseEncoder(encoder),
-		buf:         newBaseEncoder(NewBufferedEncoder()),
+		encoderBase: newBaseEncoder(encoder, opts.PolicyCompressionOptions()),
```

Comment:
Perhaps just pass `opts` to the base encoder too in case in the future we have more encoder options we need to pass along too?